user/parlib/include/parlib/signal.h

```diff
@@ -20,6 +20,7 @@ struct sigstate {
 	sigset_t mask;
 	sigset_t pending;
 	struct sigdata *data;
+        uintptr_t ss_sp;
```

Comment:
i'd also go with a different name than 'ss_sp'.  that's the name of the field in the struct sigaltstack, right?  when it's in that struct, it is the bottom of the stack.  the thing you track below in the uthread is the stacktop.  i'd just call it the sigalt_stacktop.  that way there's no confusion about what the fields are.  (and you won't need that comment down below either).